### PR TITLE
Added throttling to the load test tool and added stability test.

### PR DESCRIPTION
**Pull request by patrickkuo**
* Added stability test to the tool.
* Refactoring to allow some configuration via vm option.
* Moved NodeConnection to its own file and merged with NodeHandle.
* A teamcity task will be created to run this test.

### CODE REVIEW

**Comment by mikehearn** [comment]
Needs an update to the docsite page on how to use the tool. 

I see two types of test here - one where nodes are interfered with and another where they just self issue cash. Does the latter test p2p traffic at all? Wouldn't we be stability testing with random cash payments to other nodes, for instance?

It wasn't obvious to me how a constant load is configured. Guava has a utility that unblocks a thread at a given frequency, so you can specify that you want 20 releases/second for example. That might be easier to specify than a time based approach.

**Comment by patrickkuo** [comment]
Thanks @mikehearn.
Yes there are two type of tests, one is the existing Load test with disruption, and the second one is the stability test which execute flows at a set frequency using `Executors.scheduleAtFixedRate`, the stability test will first self issue cash and after that it will randomly pay and exit cash using `StabilityTest.crossCashTest`for P2P testing, the stability test will run without any disruption unlike the load test.
 
I will update the docsite and add more comments, will also take a look at the guava utility.

File: tools/loadtest/src/main/kotlin/net/corda/loadtest/LoadTest.kt [hunk 1]
```diff
@@ -157,53 +167,44 @@ fun runLoadTests(configuration: LoadTestConfiguration, tests: List<Pair<LoadTest
...
-            log.info("Getting node info of ${connection.hostName}")
-            val nodeInfo = connection.proxy.nodeIdentity()
-            log.info("Got node info of ${connection.hostName}: $nodeInfo!")
-            val (otherNodeInfos, nodeInfoUpdates) = connection.proxy.networkMapUpdates()
-            nodeInfoUpdates.notUsed()
-            val pubkeysString = otherNodeInfos.map {
+            log.info("Getting node info of ${connection.remoteNode.hostname}")
+            val info = connection.info
+            log.info("Got node info of ${connection.remoteNode.hostname}: $info!")
+            val (otherinfos, infoUpdates) = connection.proxy.networkMapUpdates()
```
Comment (exFalso) [inline tools/loadtest/src/main/kotlin/net/corda/loadtest/LoadTest.kt]: CamelCase 🚓

File: tools/loadtest/src/main/kotlin/net/corda/loadtest/NodeConnection.kt [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+import net.corda.nodeapi.internal.addShutdownHook
+import java.io.ByteArrayOutputStream
+import java.io.Closeable
+import java.io.OutputStream
+
+/**
+ * [NodeConnection] allows executing remote shell commands on the node as well as executing RPCs.
+ * The RPC Client start/stop must be controlled externally with [startClient] and [doWhileClientStopped]. For example
+ * if we want to do some action on the node that requires bringing down of the node we should nest it in a
+ * [doWhileClientStopped], otherwise the RPC link will be broken.
```
Comment (exFalso) [inline tools/loadtest/src/main/kotlin/net/corda/loadtest/NodeConnection.kt]: Hmm this is not true anymore, we now have autoreconnect on server bounce, could you add a TODO please to investigate whether we still need this?

**Comment by patrickkuo** [comment]
Thanks @exFalso, issues fixed

**Pull request merged**